src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -782,40 +791,68 @@ private void discoverLuckyOres()
     {
         if (luckyOres.isEmpty())
         {
+            final int defaultMineLevelName = 0;
             for (final String ore : MinecoloniesAPIProxy.getInstance().getConfig().getServer().luckyOres.get())
             {
-                final String[] split = ore.split("!");
-                if (split.length < 2)
+                final String pattern = "([\\w:]+)!(\\d+)(?:#(\\d+))?";
+                final Pattern r = Pattern.compile(pattern);
+                final Matcher m = r.matcher(ore);
+                if (!m.find())
                 {
                     Log.getLogger().warn("Wrong configured ore: " + ore);
                     continue;
                 }
 
-                final Item item = ForgeRegistries.ITEMS.getValue(new ResourceLocation(split[0]));
-                if (item == null || item == Items.AIR)
-                {
-                    Log.getLogger().warn("Invalid lucky block: " + ore);
-                    continue;
-                }
+                final String oreName = m.group(1);
+                final String chance = m.group(2);
+                final String detectedLevelName = m.group(3);
 
-                final ItemStack stack = new ItemStack(item, 1);
-                try
-                {
-                    final int rarity = Integer.parseInt(split[split.length - 1]);
-                    for (int i = 0; i < rarity; i++)
-                    {
-                        luckyOres.add(new ItemStorage(stack));
-                    }
+                String mineLevelName = defaultMineLevelName + "";
+                if (m.groupCount() == 3 && null != detectedLevelName) {
+                    mineLevelName = m.group(3);
                 }
-                catch (final NumberFormatException ex)
-                {
-                    Log.getLogger().warn("Ore has invalid rarity: " + ore);
+
+                discoverLuckyOre(ore, oreName, chance, mineLevelName);
+            }
+
+            List<ItemStorage> alternative = null;
+            int mineMaxLevel = 5;
```

Comment:
I'm not sure how to do it properly. The Configuration of the mine is in the core mod. This is in the API

Reply:
If you specifically want to fetch that value so it stays according to the actual building, you can do

```java
ModBuildings.miner.get().produceBuilding(BlockPos.ZERO, null).getMaxBuildingLevel()
```